### PR TITLE
a bit more cleanup

### PR DESCRIPTION
**Pull request by leshi**

### CODE REVIEW

File: u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/u2f/U2fAttestation.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package com.google.u2f.server.impl.attestation.u2f;
+
+import com.google.u2f.server.data.SecurityKeyData.Transports;
+import com.google.u2f.server.impl.attestation.X509ExtentionParsingUtil;
```
Comment (juanlang) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/u2f/U2fAttestation.java]: nit, not for this pull request perhaps: s/Extention/Extension/

Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/u2f/U2fAttestation.java]: ack

File: u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  @Test(expected = CertificateParsingException.class)
+  public void testMalformedCert() throws Exception {
+    U2fAttestation.Parse(TRUSTED_CERTIFICATE_MALFORMED_TRANSPORTS_EXTENSION);
+  }
+
+  @Test(expected = CertificateParsingException.class)
+  public void testValidCertNoTransports() throws Exception {
+    U2fAttestation.Parse(TRUSTED_CERTIFICATE_2);
```
Comment (juanlang) [inline u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java]: Why is this expected to throw?

Comment (leshi) [inline u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java]: There is no Transports Extension in the attestation cert and the current behavior is to throw.  I didn't want to change that behavior in this refactor.  That's part of the next change.

Comment (juanlang) [inline u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java]: Could you please add a comment to that effect? As it stands, it looks like a strong assertion that a valid certificate must contain a transports extension.

**Comment by juanlang** [comment]
lgtm modulo nit. will merge when addressed. thanks!

**Comment by leshi** [comment]
ah, I thought you wanted me to address the nit in subsequent change "not for this pull request perhaps"

**Comment by juanlang** [comment]
Sorry, different nit. I meant, please add a comment to that confusing test. Thanks!

**Comment by leshi** [comment]
done

File: u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java (hunk 2 above)
Comment (leshi) [inline u2f-ref-code/java/tests/com/google/u2f/server/impl/attestation/u2f/U2fAttestationTest.java]: done

**Pull request merged**